modules/contact/src/constraints/NormalMortarMechanicalContact.C

```diff
@@ -44,12 +44,29 @@ NormalMortarMechanicalContact::computeQpResidual(Moose::MortarType type)
       // indicates the momentum will tend to increase at this location with time, which is what we
       // want because the force vector is in the positive direction (always opposite of the
       // normals).
-      return _test_secondary[_i][_qp] * _lambda[_qp] * _normals[_qp](_component);
+      // Get the _dof_to_weighted_gap map
+
+      if (_interpolate_normals)
+        return _test_secondary[_i][_qp] * _lambda[_qp] * _normals[_qp](_component);
+      else
+      {
+        unsigned int normal_index = 0;
+        normal_index = _secondary_ip_lowerd_map.at(_i);
```

Comment:
```suggestion
        const auto normal_index = libmesh_map_find(_secondary_ip_lowerd_map, _i);
```
and everywhere else similarly